self-api/src/main/java/com/selfxdsd/api/Language.java

```diff
@@ -104,7 +128,95 @@ public final String categorize(final String command) {
      * @return String reply or null if nothing is found.
      */
     public final String reply(final String key) {
-        return this.responses.getProperty(key);
+        final String reply = this.responses.getProperty(key);
+        return tryReplyFromLink(reply);
+    }
+
+    /**
+     * Try to follow the link from reply. If the reply is not a valid
+     * link (not starting with the right protocol), it will fallback
+     * to the reply as string.
+     * <br>
+     * Supported protocols:
+     * <ul>
+     *     <li>
+     *         <b>Classpath</b> loads the file from resources:
+     *         <br>
+     *         <code>
+     *             commands.comment=classpath:replies/commands.comment_en.md
+     *         </code>
+     *     </li>
+     *     <li>
+     *         <b>File</b> loads the file from file system by absolute path:
+     *         <br>
+     *         <code>
+     *             commands.comment=file:///<full_path>/commands.comment_en.md
+     *         </code>
+     *     </li>
+     *     <li>
+     *         <b>Http(s)</b> is doing a http(s) request:
+     *         <br>
+     *         <code>
+     *             commands.comment=http://my.docs.com/commands/comment
+     *         </code>
+     *     </li>
+     * </ul>
+     *
+     * If something goes wrong like IO exception, bad request etc...,
+     * it will return null.
+     *
+     * @param linkedReply Reply as link.
+     * @return Actual reply fetched from link or null if something goes wrong.
+     * @checkstyle CyclomaticComplexity (60 lines).
+     * @checkstyle BooleanExpressionComplexity (20 lines).
+     */
+    private String tryReplyFromLink(final String linkedReply){
```

Comment:
@criske Let's name this method ``followPossibleLink``. 

Also, I like the idea of linked files for bigger responses but this method is waaay over-engineered. The ``classpath://`` option is enough, please remove the rest of the logic. 

If we will ever read language resources from a webservice, we will need some serious architectural changes.

Reply:
>The classpath:// option is enough, please remove the rest of the logic.

@amihaiemil  I agree the `file://` should be removed, classpath is enough for local files.

> If we will ever read language resources from a webservice, we will need some serious architectural changes.

@amihaiemil For this particular case, https://github.com/self-xdsd/self-docs is the webservice.
We only need to extract "Commands" listing part from https://raw.githubusercontent.com/self-xdsd/self-docs/master/projectmanager.md into a separate file, `commands.md` for ex. I think Jekyll supports including other files.
(something like `{%commands.md%}` in `projectmanager.md`)

Now on `self-pm` responses.properties we could have two entries.
```properties
commands.comment.header=Hi @%s! Here are the commands which I understand: 
commands.comment.body=https://raw.githubusercontent.com/self-xdsd/self-docs/master/commands.md
```
And then in `Commands` Conversation we merge the header with body to `SendReply`.
This way any changes on self-docs' `commands.md` will be reflected on the `commands` reply.

But until we have that refactoring on self-docs we could still use the `classpath` aproach. What do you think ? :D

Reply:
@criske I see your point but I don't agree. Self-Docs is just a static site for documentation, I don't want to mix them up together. :D 

Someone will surely end up modifying self-docs without knowing of this mechanism and break it. Please only use the classpath option.

Reply:
@criske If you want to make an integration and make sure everything is up to date, then we should generate the documentation from the code, not the other way around :D 

Maybe we can think about this later